src/Tickets/Emails/Admin/Emails_Tab.php

```diff
@@ -127,9 +146,13 @@ public function is_on_tab(): bool {
 	 * @return boolean True when on `emails` tab and on `section`
 	 */
 	public function is_on_section( $section = '' ): bool {
-		$settings = tribe( Plugin_Settings::class );
+		$current_section = tribe_get_request_var( 'section' );
+
+		if ( empty( $section ) ) {
+			return ! empty( $current_section );
+		}
 
-		return $settings->is_on_tab_section( self::$slug, $section );
+		return $section === $current_section;
```

Comment:
Why are we moving away from `is_on_tab_section`? That will make the check, we avoid code duplication, and at the same time, we're also checking against the tab and confirming that it's on the settings page.

Reply:
Oh, now I see that you use this on other functions to check if the `section` is not empty, assuming they're editing an email. Maybe we can rely on `is_editing_email ` as it already has that logic of the "empty section" and we keep this one more restricted to checking if it's on a particular section.